pkg/client/fake/client.go

```diff
@@ -1278,3 +1322,126 @@ func zero(x interface{}) {
 	res := reflect.ValueOf(x).Elem()
 	res.Set(reflect.Zero(res.Type()))
 }
+
+func extractScale(ctx context.Context, c client.Client, obj client.Object) (*autoscalingv1.Scale, error) {
+	switch obj := obj.(type) {
+	case *appsv1.Deployment:
+		var replicas int32 = 1
+		if obj.Spec.Replicas != nil {
+			replicas = *obj.Spec.Replicas
+		}
+		return &autoscalingv1.Scale{
+			ObjectMeta: obj.ObjectMeta,
```

Comment:
I think its only namespace, name, uid, resourceVersion and creationTimestamp we should copy over, not for example labels or annotations. Please update this and the corresponding tests.